tests/test_models.py

```diff
@@ -794,17 +794,97 @@ def test_delete_audit_action_ignore_calls_super(self):
+        ModelWithAuditingManager.objects.create(id=0, value="initial")
+
+        queryset = ModelWithAuditingManager.objects.all()
+        queryset.update(value='initial', audit_action=AuditAction.AUDIT)
+
+        instance, = ModelWithAuditingManager.objects.all()
+        self.assertEqual(
+            0, AuditEvent.objects.filter(object_pk=instance.pk, is_create=False,
+                                         is_delete=False).count()
+        )
+
+    def test_update_audit_action_audit_does_not_create_audit_events_if_no_audited_field_updated(self):  # noqa: E501
+        ModelWithAuditingManager.objects.create(id=0, value="initial")
+
+        queryset = ModelWithAuditingManager.objects.all()
+        queryset.update(
+            non_audited_field='updated', audit_action=AuditAction.AUDIT
+        )
+
+        instance, = ModelWithAuditingManager.objects.all()
+        self.assertEqual(instance.value, 'initial')
+        self.assertEqual(instance.non_audited_field, 'updated')
+        self.assertEqual(
+            0, AuditEvent.objects.filter(object_pk=instance.pk, is_create=False,
+                                         is_delete=False).count()
+        )
+
+    def test_update_audit_action_ignore_does_not_create_audit_events(self):
+        ModelWithAuditingManager.objects.create(id=0, value="number")
+        queryset = ModelWithAuditingManager.objects.all()
+
+        queryset.update(value='updated', audit_action=AuditAction.IGNORE)
+
+        instance, = ModelWithAuditingManager.objects.all()
+        self.assertEqual(instance.value, 'updated')
+        self.assertEqual(
+            0, AuditEvent.objects.filter(object_pk=instance.pk, is_create=False,
+                                         is_delete=False).count()
+        )
+
+    def test_update_audit_action_raise_raises_exception(self):
         queryset = AuditingQuerySet()
-        with self.assertRaises(NotImplementedError):
-            queryset.update([], audit_action=AuditAction.AUDIT)
+        with self.assertRaises(UnsetAuditActionError):
+            queryset.update(value='updated', audit_action=AuditAction.RAISE)
 
-    def test_update_audit_action_ignore_calls_super(self):
+    def test_update_audit_action_default_raises_exception(self):
         queryset = AuditingQuerySet()
-        items = object()
-        with patch.object(models.QuerySet, "update") as super_meth:
-            queryset.update(items, audit_action=AuditAction.IGNORE)
-            super_meth.assert_called_with(items)
+        with self.assertRaises(UnsetAuditActionError):
+            queryset.update(value='updated')
+
+    def test_update_audit_action_audit_rolls_back_if_fails(self):
+        ModelWithAuditingManager.objects.create(id=0, value="initial")
+
+        queryset = ModelWithAuditingManager.objects.all()
+        with patch(
+                'field_audit.models.AuditEvent.make_audit_event'
+        ) as mocked_method:
+            mocked_method.side_effect = DatabaseError()
```

Comment:
I think it would be better to raise a custom exception here rather than `DatabaseError`.  Currently, if the `super().update(...)` call raises a `DatabaseError` for some reason, this test will still pass.

Reply:
Yeah I thought about a custom exception, but I'm not sure I feel the test justification is enough. For example, I think I could assert that the `make_audit_event` method was called to fix the gap in the test you pointed out.

However thinking about this from the caller's perspective, a custom exception could provide more specific information about the auditing portion of this db write failing for whatever reason. My question is would a custom exception convey info that a DatabaseError from the AuditEvent table wouldn't?

Reply:
> I think I could assert that the `make_audit_event` method was called to fix the gap in the test you pointed out.

Yep, absolutely. That would also work.

Reply:
So I made the change to assert that the method was called, but I also changed this to raise a more generic exception, since in reality we want to test is that if make_audit_event fails for _any_ reason, roll back db changes.